store.go

```diff
@@ -397,6 +397,18 @@ type Store interface {
 	// allow ImagesByDigest to find images by their correct digests.
 	SetImageBigData(id, key string, data []byte, digestManifest func([]byte) (digest.Digest, error)) error
 
+	// ImageDirectory returns a path of a directory which the caller can
+	// use to store data, specific to the image, which the library does not
+	// directly manage.  The directory will be deleted when the image is
+	// deleted.
```

Comment:
In both cases, a small warning that the user is responsible for dealing with concurrency (including concurrency with image deletion, deleting the stored files in unpredictable order) might be valuable.